knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/branching_state/BranchingState.kt

```diff
@@ -0,0 +1,26 @@
+package com.cesar.knot_sdk.knot_state_machine.states.base_classes.branching_state
+
+import com.cesar.knot_sdk.knot_state_machine.states.base_classes.State
+import com.cesar.knot_sdk.knot_state_machine.states.base_classes.branching_state.BranchingStateActions.Timeout
+
+abstract class BranchingState : State() {
+
+    abstract override fun enter()
+
+    abstract override fun getErrorCode() : BranchingStateActions
+
+    abstract override fun getNextState(message : String) : State
+
+    /**
+     * This method returns the appropriate action given the current state
+     * conditions.
+     */
+    abstract fun getAppropriateAction() : BranchingStateActions
```

Comment:
This method's name and description are not clear enough. Please, rewrite the method's name and/or description.